packages/journey-manager/src/ui.ts

```diff
@@ -0,0 +1,380 @@
+/* eslint-disable accessor-pairs */
+import { type Client } from "@nlxai/voice-compass";
+
+/**
+ * Theme colors
+ */
+export interface ThemeColors {
+  /**
+   * Primary color
+   */
+  primary: string;
+  /**
+   * Primary color on hover
+   */
+  primaryHover: string;
+}
+
+/**
+ * Visual theme for the UI
+ */
+export interface Theme {
+  /**
+   * UI colors
+   */
+  colors: ThemeColors;
+  /**
+   * Font family
+   */
+  fontFamily: string;
+}
+
+/**
+ * Deep partial variant of the UI theme, input by the library user
+ */
+export interface PartialTheme {
```

Comment:
```typescript
export type DeepPartial<T> = {
  [P in keyof T]?: T[P] extends Array<infer I>
    ? Array<DeepPartial<I>>
    : DeepPartial<T[P]>;
};
```

Then you can just say `DeepPartial<Theme>`

Reply:
Offloading this to https://github.com/nlxai/sdk/issues/95.